src/mgmt/srv_target.c

```diff
@@ -657,3 +658,194 @@ ds_mgmt_hdlr_tgt_destroy(crt_rpc_t *td_req)
 	td_out->td_rc = rc;
 	crt_reply_send(td_req);
 }
+
+/*
+ * Return a URI string, like "ofi+sockets://192.168.1.70:44821", that callers
+ * are responsible for freeing with D_FREE.
+ */
+static int
+create_tag_uri(const char *base_uri, int i, char **uri)
+{
+	char   *type = "ofi+sockets:";
+	char   *u;
+	char   *p;
+	int	port;
+
+	/* TODO: Support other URI types. */
+	if (strncmp(base_uri, type, strlen(type)) != 0) {
+		D_ERROR("URI %s type not supported\n", base_uri);
+		return -DER_NOSYS;
+	}
+
+	/* Locate the ":" between the host and the port. */
+	p = strrchr(base_uri, ':');
```

Comment:
If the ':' between host and port does not exist, won't this catch first the one after "ofi+sockets"? In which case, won't the next line throw up a false positive and atoi() below return 0?

Reply:
Yes, this only works with correct "sockets" URIs. I discussed with Xuezhao last week. Going forward we shall change to store the URI of every context, instead of just a base URI plus a count. So this function will eventually disappear, I think.